src/util-misc.c

```diff
@@ -813,10 +809,56 @@ static int UtilMiscParseSizeStringTest02(void)
     PASS;
 }
 
+static int UtilMiscShortenStringTest01(void)
+{
+    char buffer[100];
+    const char *original = "abcdefghijklmnopqrstuvwxyz";
+    const char sep = '~';
+    const char *expected = NULL;
+
+    memset(buffer, 0, sizeof(buffer));
+    expected = "abc~xyz";
```

Comment:
(continuing From https://github.com/OISF/suricata/pull/14263#discussion_r2489191009 )

I am still surprised that 

> Shorten to 9 gives a 7 byte expected ?

@victorjulien should we add an extra byte ?

Reply:
Yeah I think here I'd expect the string to be `abc~wxyz`. The extra character would be taken from the back as that is more likely to hold relevant info in case of network device names.